src/object/srv_obj_migrate.c

```diff
@@ -2201,20 +2202,40 @@ migrate_enum_unpack_cb(struct dss_enum_unpack_io *io, void *data)
 	struct enum_unpack_arg	*arg = data;
 	struct migrate_one	*mo;
 	bool			merged = false;
+	bool			create_migrate_one = false;
+	uint32_t		parity_shard;
+	uint32_t		shard;
 	int			rc = 0;
 	int			i;
 
 	if (!daos_oclass_is_ec(&arg->oc_attr))
 		return migrate_one_create(arg, io);
 
+	/* If parity shard alive for this dkey, then ignore the data shard enumeration
+	 * from data shard.
+	 */
+	rc = obj_ec_parity_alive(arg->oh, io->ui_dkey_hash, &parity_shard);
+	if (rc < 0)
+		return rc;
+
+	shard = arg->arg->shard % obj_ec_tgt_nr(&arg->oc_attr);
+	if ((rc == 1 && io->ui_oid.id_shard != parity_shard) || io->ui_oid.id_shard == shard) {
+		D_DEBUG(DB_REBUILD, DF_UOID" ignore shard "DF_KEY"/%u.\n",
+			DP_UOID(io->ui_oid), DP_KEY(&io->ui_dkey), shard);
+ 		return 0;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs